stdlib/public/SwiftShims/MetadataSections.h

```diff
@@ -0,0 +1,55 @@
+//===--- RuntimeShims.h - Access to runtime facilities for the core -------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef SWIFT_STDLIB_SHIMS_METADATASECTIONS_H
+#define SWIFT_STDLIB_SHIMS_METADATASECTIONS_H
+
+#include "SwiftStddef.h"
+#include "SwiftStdint.h"
+
+#ifdef __cplusplus
+namespace swift { extern "C" {
+#endif
+
+struct MetadataSectionRange {
+  __swift_uintptr_t start;
+  __swift_size_t length;
+};
+
+struct MetadataSections {
+  __swift_uintptr_t version;
+  __swift_uintptr_t reserved;
+
+  struct MetadataSections *next;
+  struct MetadataSections *prev;
```

Comment:
You can use the C++ mutable keyword for these two fields. This tells the compiler that it's OK to mutate these two fields within a const MetadataSections instance.

Reply:
I can't because this has to be a plain C struct, in order to be accessible from swift :(